swagger-generator/swagger-generator.ts

```diff
@@ -0,0 +1,238 @@
+    const method = routeData.method;
+
+    let newRoute;
+    if (method.toLowerCase() === 'get') {
+      newRoute = swagger.get(route);
+    } else if (method.toLowerCase() === 'post') {
+      newRoute = swagger.post(route);
+    } else if (method.toLowerCase() === 'put') {
+      newRoute = swagger.put(route);
+    } else if (method.toLowerCase() === 'patch') {
+      newRoute = swagger.patch(route);
+    } else if (method.toLowerCase() === 'head') {
+      newRoute = swagger.head(route);
+    } else if (method.toLowerCase() === 'options') {
+      newRoute = swagger.options(route);
+    } else if (method.toLowerCase() === 'delete') {
+      newRoute = swagger.delete(route);
+    }
+
+    newRoute.parameters(parameters)
+      .operationId(id)
+      .tag(tag)
+      .summary(summary)
+      .response(200);
+  }
+
+  fs.writeFileSync('swagger.json', JSON.stringify(swagger.doc));
+}
+
+function getSwaggerParametersForRoute(route): Array<SwaggerParameter> {
+  const swaggerParameters: Array<SwaggerParameter> = [];
+
+  const parametersInRoute: Array<string> = extractParametersFromRoute(route);
+
+  for (const parameter of parametersInRoute) {
+    swaggerParameters.push(getSwaggerParameterByPropertyName(parameter));
+  }
+
+  return swaggerParameters;
+}
+
+function extractParametersFromRoute(route: string): Array<string> {
+  const parameters: Array<string> = route.split('/')
+    .filter((routePart: string): boolean => {
+      return routePart.startsWith(':');
+    })
+    .map((parameter: string): string => {
+      return parameter.replace(':', '');
+    });
+
+  return parameters;
+}
+
+function createSwaggerRoutes(sourceFile: typescript.SourceFile, properties: Array<any>): void {
+  for (const property of properties) {
+    const propertyTextWithComment: string = property.getFullText(sourceFile);
+    const propertyTextWithoutComment: string = property.getText(sourceFile);
+    const propertyDocumentation: string = propertyTextWithComment.replace(propertyTextWithoutComment, '');
+
+    const routeName: string = property.name.getText(sourceFile);
+    const swaggerRoute: SwaggerRoute = convertPropertyDocumentationToSwaggerRoute(propertyDocumentation);
+
+    addSwaggerRoute(routeName, swaggerRoute);
+  }
+}
+
+function extractSwaggerDataFromContracts(): void {
+  const restSettingsFileName = 'node_modules/@process-engine/consumer_api_contracts/src/rest_settings.ts';
+
+  const program: typescript.Program = typescript.createProgram([restSettingsFileName], {});
+  const sourceFile: typescript.SourceFile = program.getSourceFile(restSettingsFileName);
+
+  for (const statement of sourceFile.statements) {
+    statement.forEachChild((statementNode: any): void => {
+      if (
+        statementNode.declarations === undefined ||
+        statementNode.declarations.length < 1 ||
+        !typescript.isVariableDeclaration(statementNode.declarations[0])) {
+
+        return;
+      }
+
+      const variable: any = statementNode.declarations[0];
```

Comment:
```suggestion
      const variable = statementNode.declarations[0];
```

Reply:
Ok.

Dann mach aber besser:

```suggestion
      const variable = statementNode.declarations[0] as any;
```